core/store/orm/orm.go

```diff
@@ -326,6 +329,47 @@ func (orm *ORM) LinkEarnedFor(spec *models.JobSpec) (*assets.Link, error) {
 	return earned, nil
 }
 
+// UpsertErrorFor upserts a JobSpecError record, incrementing the occurances counter by 1
+// if the record is found
+func (orm *ORM) UpsertErrorFor(jobID *models.ID, description string) error {
+	orm.MustEnsureAdvisoryLock()
```

Comment:
I wouldn't be doing my job if I didn't ask why we need an advisory lock here ;)

See: https://www.notion.so/chainlink/Horizontal-scalability-f65082317188424f881b65b5c1443541

Reply:
Now that it's combined into one query I don't think we do. But would like to discuss!